src/components/Pinpad/Pinpad.tsx

```diff
@@ -0,0 +1,100 @@
+import { useState, useEffect } from "react";
+import { ErrorDiv } from "../ErrorDiv/ErrorDiv";
+
+import styles from "./Pinpad.module.css";
+import {
+	PINPAD_MUI_KEYS,
+	LINK_ID_LENGTH,
+	VITESSCE_LINK_SITE,
+} from "../../utils/constants";
+import { exampleConfigHeadset } from "../../utils/config-examples";
+import { PinpadKey } from "../Pinpadkey";
+
+type ErrorType = string | null;
+
+export const Pinpad = () => {
+	const [linkId, setLinkId] = useState<string>("");
+	const [error, setError] = useState<ErrorType>(null);
+
+	const handleKeyPress = (key: string) => {
+		if (key === PINPAD_MUI_KEYS.BACKSPACE) {
+			setLinkId(() => linkId.slice(0, -1));
+		} else if (key === PINPAD_MUI_KEYS.DONE) {
+			attemptLogin();
+		} else if (linkId.length < LINK_ID_LENGTH && !isNaN(Number(key))) {
+			setLinkId(() => linkId + key);
+		}
+	};
```

Comment:
Same as below, we can drop the `() =>`:

```suggestion
	const handleKeyPress = (key: string) => {
		if (key === PINPAD_MUI_KEYS.BACKSPACE) {
			setLinkId(linkId.slice(0, -1));
		} else if (key === PINPAD_MUI_KEYS.DONE) {
			attemptLogin();
		} else if (linkId.length < LINK_ID_LENGTH && !isNaN(Number(key))) {
			setLinkId(linkId + key);
		}
	};
```

---

Alternatively, this could also be refactored to remove the function's direct dependency on `linkId` with the function setter approach, which would be useful if we end up needing to wrap it in `useCallback` for performance reasons, but that would also require making `attemptLogin` a callback as well.

```suggestion
	const handleKeyPress = (key: string) => {
		if (key === PINPAD_MUI_KEYS.DONE) {
			attemptLogin();
		} else {
			setLinkId((prevLinkId) => {
				if (key === PINPAD_MUI_KEYS.BACKSPACE) {
					return prevLinkId.slice(0, -1);
				}
				if (prevLinkId.length < LINK_ID_LENGTH && !isNaN(Number(key))) {
					return prevLinkId + key;
				}
				return prevLinkId;
			})
		}
	};
```

---

To be clear: the first suggestion is probably sufficient/easier to follow, the second suggestion is more performant for memoization if that becomes a concern down the line.